### ISSUE TITLE
Logical byte operations

### ISSUE THREAD

**Issue by jackfirth** [jackfirth/rebellion#166]
The [`rebellion/binary/byte`](https://docs.racket-lang.org/rebellion/Bytes.html) module could use some functions for performing logical AND, OR, NOT, etc. operations on the eight bits of a byte. Specifications:

- [ ] `(byte-and [left byte?] [right byte?]) -> byte?` — Logical AND
- [ ] `(byte-or [left byte?] [right byte?]) -> byte?` — Logical OR
- [ ] `(byte-not [b byte?]) -> byte?` — Logical NOT
- [ ] `(byte-xor [left byte?] [right byte?]) -> byte?` — Exclusive OR (like OR, but if both bits are 1 the output bit is 0)
- [ ] `(byte-nand [left byte?] [right byte?]) -> byte?` — Negated AND, equivalent to `(byte-not (byte-and left right))`
- [ ] `(byte-nor [left byte?] [right byte?]) -> byte?` — Negated OR, equivalent to `(byte-not (byte-or left right))`

These functions _could_ take multiple arguments instead of only two, but I think [reducers](https://docs.racket-lang.org/rebellion/Reducers.html) make that somewhat unnecessary. For example, a variable-arity `(byte-or x y z ...)` would be equivalent to `(reduce (make-fold-reducer byte-or 0) x y z ...)`, and the reducer approach would work with`bytes?` values, infinite streams of bytes, and input byte ports.

**Comment by jackfirth** [comment]
Examples:

```racket
> (byte-and 0 0)
0
> (byte-and 0 255)
0
> (byte-or 0 255)
255
> (byte-and 255 255)
255
> (byte-and (byte 1 1 1 1 0 0 0 0) (byte 0 0 0 0 1 1 1 1))
0
> (byte-or (byte 1 1 1 1 0 0 0 0) (byte 0 0 0 0 1 1 1 1))
255
```

**Comment by jackfirth** [comment]
Closed by #225.
